src/vm/class.cpp

```diff
@@ -603,12 +603,17 @@ HRESULT EEClass::AddMethod(MethodTable * pMT, mdMethodDef methodDef, RVA newRVA,
     // Get the new MethodDesc (Note: The method desc memory is zero initialized)
     MethodDesc *pNewMD = pChunk->GetFirstMethodDesc();
 
-    ACQUIRE_STACKING_ALLOCATOR(pStackingAllocator);
 
     // Initialize the new MethodDesc
+    // The stacking allocator being passed here as NULL replaces a silent optimization bug that was being masked before here.
+    // This was reported in https://github.com/dotnet/coreclr/issues/26016, and was caused by PR https://github.com/dotnet/coreclr/pull/24177
+    // Previously, &GetThread()->m_MarshalAlloc was being optimized into an offset from a globl thread-static. After the PR, this resulted
+    // in a field access on the thread object, but there's no Thread object as this function gets called on the RC Thread.
+    // This is not a fix, but rather a workaround to get to the state prior to the cusomer scenario regression.
+    // Any case in which the builder requires to use the allocator with AV, which is consistent with the long-existing behavior.
     MethodTableBuilder builder(pMT,
```

Comment:
Is it really that hard to implement a fix that is correct and does not just try to preserve the previous mine-field?

What about something like this?

```
    // This method runs on a debugger thread. Debugger threads do not have Thread object that caches StackingAllocator. Use a a local StackingAllocator instead.
    StackingAllocator stackingAllocator;

    MethodTableBuilder builder(pMT,
                                                  pClass,
                                                  &stackingAllocator,
...
```

Reply:
@jkotas I guess I misunderstood the usage of the `StackingAllocator`. If nothing that gets constructed will outlast the builder by hanging off the method table/method desc, I definitely prefer this change.